### DISCUSSION TITLE
subscription not receiving updates

### DISCUSSION THREAD

**Discussion by Conner05** [supabase/supabase#865] [Questions] [answered]
I am using the Next.js and Stripe starter kit for my project. I am trying to subscribe to changes to my users table. I get nothing from the `console.log()` I also get nothing when I change my `.from(...)` to `.from('*')`. Also, I ran this query as suggested in another similar post: `select * from pg_replication_slots;` and am seeing the same thing they did (`slotname` is `realtime` and `plugin` is `pgoutput`. So it looks like everything is setup fine for realtime. Any ideas what's going on? 

This is the code running the update. I know it works b/c it actually changes the `full_name` property of my user.

```
export const updateUserName = async (user, name) => {
  await supabase
    .from('users')
    .update({
      full_name: name
    })
    .eq('id', user.id)
}
```

This is the code used to set up the subscription

````
const subscribeToUserUpdates = (userId) => {
    return supabase
      .from(`users:id=eq.${userId}`)
      .on('*', (res) => {
        console.log('res', res);
        userUpdateEvents.forEach((event) => event?.run());
      })
      .subscribe();
  };

  useEffect(() => {
    const session = supabase.auth.session();
    setSession(session);
    setUser(session?.user ?? null);
    let userListener = subscribeToUserUpdates(session?.user?.id ?? null);
    console.log(`subscribed to user updates for id ${session?.user.id}`);
    const { data: authListener } = supabase.auth.onAuthStateChange(
      async (event, session) => {
        setSession(session);
        setUser(session?.user ?? null);
        userListener = subscribeToUserUpdates(session?.user?.id ?? null);
        console.log(
          `subscribed to user updates for id ${session?.user.id} in auth state change`
        );
      }
    );

    return () => {
      authListener.unsubscribe();
      userListener.unsubscribe();
    };
  }, []);

**Comment by inian** [comment]
Can you email beta@supabase.io with the project id @Conner05? Thanks!

**Reply by Conner05** [reply]
yep!

**Comment by soedirgo** [comment, marked as answer]
Hey @Conner05, can you go to Database > Replication > `supabase_realtime` > Source and see if toggling the `users` table solves the issue?

**Reply by Conner05** [reply]
@soedirgo that did _something_. Now, if I change ```.from(`users:id=eq.${userId}`)``` to ```.from(`*`)``` then it works. Interestingly enough, the users table had replication toggled off by default. I suppose that's probably managed from the migration, though - not sure.

Anyway, now I just need to figure out why the row level specification isn't working.

**Reply by Conner05** [reply]
I got it! Turns out I needed to remove the `''` from the `userId` guid - I thought it needed to be passed as a string.

So, I guess it just needed to be turned on!

**Reply by soedirgo** [reply]
Glad that helped! @kiwicopple should we update [this line](https://github.com/vercel/nextjs-subscription-payments/blob/5953c9ae34f0b2e7ba368a4a671dabb1c78cd2f3/schema.sql#L145) to include the `users` table?
